src/upnp_device.c

```diff
@@ -71,15 +74,19 @@ int upnp_add_response(struct action_event *event,
 		return -1;
 	}
 
+	actionResult = UpnpActionRequest_get_ActionResult(event->request);
```

Comment:
Prefetching the ActionResult here seems to be causing at least one problem as reported by @ashkitten.

Moving `UpnpActionRequest_get_ActionResult` inline with `UpnpAddToActionResponse` resolved the issue.

Why? Beats me.

Reply:
Looks like #170 handles this by calling `UpnpActionRequest_set_ActionResult` before returning.

Since we've made a copy of the pointer to ActionResult, (and possibly modified it) )we need to update the ActionResult in the event "class".